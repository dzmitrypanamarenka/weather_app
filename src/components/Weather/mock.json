{
  "forecastData": {
    "weather": "fine",
    "name": "random",
    "main": {
      "temp": 1,
      "humidity": 1,
      "pressure": 1
    },
    "sys": {
      "country": "DE"
    },
    "wind": {
      "speed": 1
    }
  }
}
