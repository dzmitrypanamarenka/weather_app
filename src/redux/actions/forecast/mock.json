{
  "base": "stations",
  "clouds": {
    "all": 40
  },
  "cod": 200,
  "coord": {
    "lat": 52.52,
    "lon": 13.4
  },
  "dt": 1524669600,
  "id": 6545310,
  "main": {
    "humidity": 50,
    "pressure": 1008,
    "temp": 14.51,
    "temp_max": 15,
    "temp_min": 14
  },
  "name": "Berlin Mitte",
  "sys": {
    "country": "DE",
    "id": 4892,
    "message": 0.0029,
    "sunrise": 1524627968,
    "sunset": 1524680612,
    "type": 1
  },
  "weather": [
    {
      "description": "scattered clouds",
      "icon": "03d",
      "id": 802,
      "main": "Clouds"
    }
  ],
  "wind": {
    "deg": 290,
    "gust": 14.9,
    "speed": 8.7
  }
}
