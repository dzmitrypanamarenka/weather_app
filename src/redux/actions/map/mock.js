export const mockGeolocationResolve = {
  getCurrentPosition: jest.fn((res) => (
    res({
      coords: {
        latitude: 52.520007,
        longitude: 13.404954,
      }
    })
  ))
};

export const mockGeolocationReject = {
  getCurrentPosition: jest.fn((res, rej) => rej(new Error()))
};
