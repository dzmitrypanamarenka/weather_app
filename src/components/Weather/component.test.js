import React from 'react';
import renderer from 'react-test-renderer';

import { Weather } from '../';
import mock from './mock';

describe('Snapshot test', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(<Weather forecast={mock}/>)
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});

// describe('Component render', () => {
//   let wrapper;
//
//   beforeEach(() => {
//     wrapper = mount(LocationWithStore);
//   });
//
//   it('should render Location component', () => {
//     expect(wrapper.find('.container').length)
//       .toEqual(1);
//   });
//   it('should contains mounted components', () => {
//     expect(wrapper.containsAllMatchingElements([
//       Map,
//       Marker,
//       InfoWindow,
//       Message,
//     ]))
//       .toEqual(true);
//   });
// });
