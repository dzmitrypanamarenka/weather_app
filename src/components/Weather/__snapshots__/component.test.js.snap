// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should match snapshot 1`] = `
<div
  className="weather-container"
>
  <div
    className="blur-wrap"
  >
    <div
      className="forecast-wrap jumbotron"
    >
      <h2
        className="title -place"
      >
        random
        ,
        DE
      </h2>
      <h1
        className="title -temp"
      >
        1
        Â°C
      </h1>
      <img
        alt=""
        className="image"
        src="http://openweathermap.org/themes/openweathermap/assets/vendor/owm/img/widgets/undefined.png"
      />
      <ul
        className="list"
      >
        <li
          className="item"
        >
          <span
            className="title"
          >
            Humidity:
          </span>
          <span
            className="title -data"
          >
            1
            %
          </span>
        </li>
        <li
          className="item"
        >
          <span
            className="title"
          >
            Pressure:
          </span>
          <span
            className="title -data"
          >
            1
            hPa
          </span>
        </li>
        <li
          className="item"
        >
          <span
            className="title"
          >
            Wind:
          </span>
          <span
            className="title -data"
          >
            1
            m/s
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
